Code/Framework/AzCore/Tests/Math/Vector2Tests.cpp

```diff
@@ -7,9 +7,8 @@
  */
 
 #include <AzCore/Math/MathUtils.h>
-#include <AzCore/Math/Vector3.h>
-#include <AzCore/Math/Vector4.h>
 #include <AzCore/Math/Vector2.h>
+#include <AzCore/Math/Vector4.h>
 #include <AzCore/UnitTest/TestTypes.h>
 #include <AZTestShared/Math/MathTestHelpers.h>
 #include <AzCore/Math/SimdMath.h>
```

Comment:
Sounds like we have a circular include issue if there a compilation issue related to moving the SimdMath.h include to the top of this file.

Reply:
SimdMath.h does not include the symbols for the intrinsic values not sure what the include its looking for :(

Reply:
It doesn't seem to know what these types are: 

```
/data/workspace/o3de/Code/Framework/AzCore/./AzCore/Math/SimdMathVec1.h:20:31: error: unknown type name '__m128'

[2022-07-25T15:36:49.653Z]             using FloatType = __m128;

[2022-07-25T15:36:49.653Z]                               ^

[2022-07-25T15:36:49.653Z] /data/workspace/o3de/Code/Framework/AzCore/./AzCore/Math/SimdMathVec1.h:21:31: error: unknown type name '__m128i'

[2022-07-25T15:36:49.653Z]             using Int32Type = __m128i;

[2022-07-25T15:36:49.653Z]                               ^
....
```
there getting included in one of those other includes above or multiples not sure which one. seems like SimdMath.h is missing an include

Reply:
I added the missing includes into the individual SimdMath*.h so this condition shouldn't happen.